datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -257,7 +257,56 @@ impl PhysicalExpr for BinaryExpr {
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
         use arrow::compute::kernels::numeric::*;
 
+        #[inline]
+        fn check_short_circuit(arg: &ColumnarValue, op: &Operator) -> bool {
+            let data_type = arg.data_type();
+            if data_type == DataType::Boolean {
```

Comment:
I think this read easier by using `match (arg, *op)` in combination with `if` guard.

```
match (arg, *op) {
    (ColumnarValue::Array(array), Operator::And) => {
       ...
}
```

Reply:
ok,I'll fix it, thanks for the review.